ui/raidboss/data/07-dt/hunts/shaaloani.ts

```diff
@@ -434,6 +565,249 @@ const triggerSet: TriggerSet<Data> = {
+            dir1 = 'front';
+          }
+
+          if (safeSpot & backFrontSpot & TtokSafeSpots.Right) {
+            dir2 = 'right';
+          } else if (safeSpot & backFrontSpot & TtokSafeSpots.Left) {
+            dir2 = 'left';
+          }
+          data.ttokSandOrbOnSet++;
+          data.ttokRotated = rotation;
+          return {
+            alertText: output.triple!({
+              inOut: output.outOfHitbox!(),
+              dir2: output[dir1]!(),
+              dir3: output[dir2]!(),
+            }),
+          };
+        }
+        return {
+          infoText: output.awayFrom!({ out: output.outOfHitbox!(), dir: awaySide }),
+        };
+      },
+    },
+    // The boss does either in; out; in-right + out-left; or out-right + in-left
+    // can be combined with sand orbs explosions.
+    {
+      id: 'Hunt Ttokrrone Desert Tempest',
+      type: 'StartsUsing',
+      netRegex: { id: ttokrroneDesertTempestIds, source: 'Ttokrrone', capture: true },
+      // condition: (data) => data.inCombat,
+      durationSeconds: 7,
+      alertText: (data, matches, output) => {
+        const tempest = ttokrroneDesertTempest[matches.id];
+        if (tempest === undefined)
+          throw new UnreachableCode();
+        const pattern = data.ttokSandOrbPatterns[data.ttokSandOrbOnSet];
+        let inOut: TtokrroneDodge = 'unknown';
+        let rightLeft: TtokrroneDodge | null = null;
+        let backFront: TtokrroneDodge | null = null;
+        let backFrontSpot = TtokSafeSpots.All;
+
+        if (tempest & TtokSafeSpots.In) {
+          inOut = 'in';
+        } else if (tempest & TtokSafeSpots.Out) {
+          inOut = 'out';
+        }
+        if (tempest === TtokSafeSpots.InRight) {
+          rightLeft = 'right';
+        } else if (tempest === TtokSafeSpots.InLeft) {
+          rightLeft = 'left';
+        }
+
+        // if the boss rotates we give up and only call out the Tempest portion.
+        if (pattern && data.ttokRotated === 0) {
+          const safeSpot = tempest & pattern;
+
+          if (safeSpot & TtokSafeSpots.Back) {
+            backFrontSpot = TtokSafeSpots.Back;
+            backFront = 'back';
+          } else if (safeSpot & TtokSafeSpots.Front) {
+            backFrontSpot = TtokSafeSpots.Front;
+            backFront = 'front';
+          }
+          // if the tempest was only in or out, but the sand pattern needs right or left
+          if (rightLeft === null && !(safeSpot & TtokSafeSpots.Middle)) {
+            // we also need to check the other selected direction (front/back)
+            if (safeSpot & backFrontSpot & TtokSafeSpots.Right) {
+              rightLeft = 'right';
+            } else if (safeSpot & backFrontSpot & TtokSafeSpots.Left) {
+              rightLeft = 'left';
+            }
+          }
+          if (rightLeft === null) {
+            // swap to cover if tempest is in/out and sand orbs are all front/back
+            [rightLeft, backFront] = [backFront, rightLeft];
```

Comment:
Hmm - not sure I understand the need for the swap here, since you're explicitly referencing `rightLeft` and `backFront` in the calls to `output.triple!()` and `output.double!()`.  Sorry if I'm missing something obvious - could you explain a little more?

Reply:
I think I originally fixed a bug here since this is where I was looking for problems. But I do think it makes sense to just add another if statement at the end to clear what is happening (this is what I was doing with the swap).